src/System.Private.Xml/tests/XmlReader/XmlResolver/XmlSystemPathResolverTests.cs

```diff
@@ -29,6 +32,7 @@ public static void TestResolveRelativePaths()
                 if (shouldDelete)
                 {
                     File.Delete(path);
+                    Directory.SetCurrentDirectory(curDir);
```

Comment:
Why do you do SetcurrentDirectory only if `shouldDelete`?  
I actually don't know why `shouldDelete` exists. Hopefully our random paths don't collide. They have ~50^(8+3) possible variation.s